### DISCUSSION TITLE
Elementary OS on  Yoga Slim 9i

### DISCUSSION THREAD

**Discussion by marika144** [elementary/triage#177] [Q&A]
I recently got an Yoga  Slim 9i and want to use it with Elementary OS. I installed it and  it works fine,  but there are some issues.

The screen tends to flicker and the keyboard is unresponsive for a few minutes (sometimes it works right away, but usually 3 - 5 minutes) after turning the laptop on. It might be a problem caused by the drivers, because it worked just fine when  I used Windows 10. I'm quite certain that the hardware is not the problem. 

I have no experience in solving problems like this and would be very thankful for any ideas!

**Comment by vjr** [comment]
For the keyboard issue, try specifying kernel boot parameter `i8042.nopnp=1` which works for my HP Pavilion 14 tiger lake platform laptop. If this doesn't work for you , look up other settings for the `i8042` module in the kernel parameters documentation online?

For the screen flicker - what graphics hardware do you have? (either iGPU or dGPU) - if you have an nvidia you can try the `nomodeset` kernel boot option and install the latest drivers relevant to your graphics.

**Reply by marika144** [reply]
`i8042.nopnp=1` did  nothing for me but disable the touch, but I tried the combination of `i8042.nopnp=1`and `i8042.dumbkdb=1`. That fixed the keyboard  issue but the touch was still dead. So I tried only using `i8042.dumbkdb=1` and that worked for keyboard and touch! Thank you!

My GPU is Intel® Iris® Xe Graphics, so no nvidia..

**Reply by vwtyler** [reply]
I have the Lenovo IdeaPad S940-14IWL with Intel UHD Graphics 620 (WhiskeyLake-U GT2) and the `nomodeset` worked for me:

```
sudo nano /etc/default/grub

#Find the line starting with GRUB_CMDLINE_LINUX_DEFAULT and add nomodeset
GRUB_CMDLINE_LINUX_DEFAULT="quiet splash nomodeset"

#Save and exit 

#Update GRUB 
sudo update-grub

#Reboot
sudo reboot
```
